cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -385,6 +374,9 @@ private String rebalanceStateConditionType(KafkaRebalanceStatus status) {
 
         return computeNextStatus(reconciliation, host, apiClient, kafkaRebalance, currentState, rebalanceAnnotation, rebalanceOptionsBuilder)
            .compose(desiredStatusAndMap -> {
+               if (Annotations.isReconciliationPausedWithAnnotation(kafkaRebalance)) {
+                   return updateStatus(reconciliation, kafkaRebalance, desiredStatusAndMap.status, null).compose(i -> Future.succeededFuture());
+               }
```

Comment:
The reconciliation being paused is not question of the status only. The Operator should not do anything when it is paused. So does it really make sense to do this only after the `computeNextStatus` etc.?

Reply:
As I said before and as Jakub points out, we need to avoid doing any work we don't need too when in the paused state. So we should avoid triggering `computeNextStatus` if we are paused. 

I think the state was set to New when pauses so that, when the pause annotation was removed, the Kafka Rebalance CR would be reconciled automatically without any further intervention.

Reply:
The issue is that once the reconciliation is unpaused --> we stay in the `New` state and I guess no matter on which state we pause our resource will always continue form `Proposal Ready` state

Reply:
This is because one the resource is unpaused then the current state is `New` and then the next steps will happen accordingly